### PR TITLE
Migrate `collection attributes` store to Pinia

### PR DESCRIPTION
**Pull request by davelopez**
Includes some unit tests for the store.

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/stores/collectionAttributesStore.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+    async function fetchAttributes(params: { hdcaId: string }) {
+        Vue.set(loadingAttributes.value, params.hdcaId, true);
+        try {
+            const attributes = await Service.fetchCollectionAttributes(params);
+            Vue.set(storedAttributes.value, params.hdcaId, attributes);
+            return attributes;
+        } finally {
+            Vue.delete(loadingAttributes.value, params.hdcaId);
```
Comment (ElectronicBlueberry) [inline client/src/stores/collectionAttributesStore.ts]: With the composition api these functions can be imported directly, avoiding the default import.

```suggestion
            set(storedAttributes.value, params.hdcaId, attributes);
            return attributes;
        } finally {
            del(loadingAttributes.value, params.hdcaId);
```

File: client/src/stores/collectionAttributesStore.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const isLoadingAttributes = computed(() => {
+        return (hdcaId: string) => {
+            return loadingAttributes.value[hdcaId] ?? false;
+        };
+    });
+
+    async function fetchAttributes(params: { hdcaId: string }) {
+        Vue.set(loadingAttributes.value, params.hdcaId, true);
+        try {
+            const attributes = await Service.fetchCollectionAttributes(params);
```
Comment (ElectronicBlueberry) [inline client/src/stores/collectionAttributesStore.ts]: Only importing the required functions from a module can improve tree-shaking.

```suggestion
            const attributes = await fetchCollectionAttributes(params);
```

File: client/src/stores/collectionAttributesStore.ts (hunk 1 above)
Comment (davelopez) [reply client/src/stores/collectionAttributesStore.ts]: Done!

File: client/src/stores/collectionAttributesStore.ts (hunk 2 above)
Comment (davelopez) [reply client/src/stores/collectionAttributesStore.ts]: Good point! Thanks!

**Review by dannon — APPROVED** [review]

**Comment by dannon** [comment]
Thanks!

**Pull request merged**